src/python/py/models/builder.py

```diff
@@ -1839,7 +1961,14 @@ def make_model(self, input_path):
             elif module.__class__.__name__.endswith("DecoderLayer") and self.layer_id < self.num_layers:
                 # Each decoder layer of model
                 print(f"Reading decoder layer {self.layer_id}")
-                self.make_layer(self.layer_id, module)
+
+                # Some model stores rotary_emb at model class instead of decoder layer.
+                model_rotary_emb = None
+                if hasattr(model, "model"):
+                    if hasattr(model.model, "rotary_emb"):
+                        model_rotary_emb = model.model.rotary_emb
```

Comment:
Let's define this in the `Nemotron` class instead of using `kwargs` in `make_layer`

Reply:
I didn't get you; the 'model' object is created & destroyed locally in make_model() function call. So, accessing 'model.model.rotary_emb' in Nemotron is not possible or I am missing something?

Reply:
In the `Nemotron` class, you can override the `make_attention` method to add the `rotary_emb` attribute (i.e. `attention.rotary_emb = model.model.rotary_emb`) and then call the parent method. Here's an example.

https://github.com/microsoft/onnxruntime-genai/blob/25636b434bfb93564be304b3218931a68fb6ac9f/src/python/py/models/builder.py#L2646-L2651

Then, the `kwargs` and `model_rotary_emb` changes can be reverted.

Reply:
That makes sense, I have fixed the code now.